### PR TITLE
refactor: Modernize codebase

### PR DESCRIPTION
**Pull request by MichaelDeBoey**

### CODE REVIEW

File: src/createBannedAttributeRule.js [hunk 1]
```diff
@@ -1,114 +1,113 @@
...
       ? preferred
       : negatedPreferred;
-  }
 
   const isBannedArg = (node) =>
     attributes.some((attr) => attr === node.arguments[0].value);
 
   return {
-    [`CallExpression[callee.property.name=/${preferred}|${negatedPreferred}/][callee.object.property.name='not'][callee.object.object.callee.name='expect']`](
+    [`CallExpression[callee.property.name=/${preferred}|${negatedPreferred}/][callee.object.property.name='not'][callee.object.object.callee.name='expect']`]: (
```
Comment (benmonro) [inline src/createBannedAttributeRule.js:14]: why is this `:` needed here?

Comment (benmonro) [inline src/createBannedAttributeRule.js:14]: I actually prefer this syntax over the arrow syntax in this case.  can we revert those please?

Comment (MichaelDeBoey) [reply src/createBannedAttributeRule.js:14]: I’ve done it in order to not violate the max-line linting error so eassily

Comment (benmonro) [reply src/createBannedAttributeRule.js:14]: Ok then it probably doesn't make sense for a lint rule repo

Comment (MichaelDeBoey) [reply src/createBannedAttributeRule.js:14]: I think it does, so we only disable it when really necessary

Comment (benmonro) [reply src/createBannedAttributeRule.js:14]: hmm alternatively did you try breaking the string into multiple lines?

maybe this works?
```js
[`
CallExpression[callee.property.name=/${preferred}|${negatedPreferred}/]
[callee.object.property.name='not']
[callee.object.object.callee.name='expect']
`](
```

Comment (MichaelDeBoey) [reply src/createBannedAttributeRule.js:14]: @benmonro Breaking the string into multiple lines will make the tests fail

Comment (benmonro) [reply src/createBannedAttributeRule.js:14]: gotcha.  ok then we need to disable the rule.  A lint plugin that uses the selector syntax is going to have to disable quite a few of them so it probably makes sense to either change it for the rules directory or something, but I really want to keep this syntax.  it's probably my favorite feature of writing a lint rule. :)

Comment (MichaelDeBoey) [reply src/createBannedAttributeRule.js:14]: We currently only disable the `max-lines-per-function` rule in 1 file

Comment (benmonro) [reply src/createBannedAttributeRule.js:14]: alright.  do you mind reverting these back to not use arrow functions then?  In these files I much prefer the object function syntax.

Comment (MichaelDeBoey) [reply src/createBannedAttributeRule.js:14]: I'm on it 😉

**Pull request merged**